### ISSUE TITLE
Load i18n labels from options

### ISSUE THREAD

**Issue by soymanuvalle** [PDFTron/webviewer-ui#564]
Hello, we're using the Webviewer as an "external app" and only making modifications to the initializer options, etc.

We have faced with an issue related to the i18n labels. As the webviewer is being used externally, we already have a localization system on our main app, so we would like to load the webviewer labels from our existing data. 

I18next has a feature for doing this, which is named **resources**. I would like to know if we could submit a pull request for this.

We have thought on doing a modification to the file `src/helpers/setupI18n.js`:
```[javascript]
if (state.advanced.disableI18n) {
    i18next.init(options, callback);
} else if (state.advanced.i18nResources) {
    i18next.use(XHR).init({
            ...options,
            resources: state.advanced.i18nResources,
        },
        callback
    );
} else {
    i18next.use(XHR).init({
            ...options,
            backend: {
                loadPath: "./i18n/{{ns}}-{{lng}}.json",
            },
        },
        callback
    );
}
```

So we could initialize our web reader like this:
```
WebViewer({
        path: '/WebViewer',
        i18nResources: {
            en: {
                translation: {
                    "action": {
                        "apply": "Apply text"
                    }
                    // ...pdfTron labels
                }
            },
            es: {
                translation: {
                    "action": {
                        "apply": "Texto aplicar"
                    }
                    // ...pdfTron labels
                }
            }
        }
    },
    viewer.current
);
```

To do so we would also add the default option value on `src/redux/initialState.js`:
```
advanced: {
    // ...
    i18nResources: null,
}
```

Thanks

**Comment by ZhijieZhang** [comment]
Hi,

First of all, thanks for the PR and your detailed issue contents.

I've reviewed your PR, and it looks good to me. One thing I'd like to mentions is that putting `i18nResources` in `src/redux/initialState.js` doesn't yet make it read the translation you pass to the WebViewer constructor.

To make this work as expected we would also need to make a change to `webviewer.js`, which is the file that makes the iframe element, reads the options in the WebViewer constructor, and append them as a part of the url to initialize the UI. That's being said, we would need to do this instead:
```javascript
advanced : {
  // ...
  i18nResources: getHashParams('i18nResources', null)
}
```

It looks like what you wanted to do can be achieved with the current implementation already, though it's not documented properly.

WebViewer actually takes in a `disabledI18n` option, which will set the i18next to not fetch the existing translation json files.
https://github.com/PDFTron/webviewer-ui/blob/9da862531ab63168497a1d26b41d235d092c5e1b/src/redux/initialState.js#L249
https://github.com/PDFTron/webviewer-ui/blob/9da862531ab63168497a1d26b41d235d092c5e1b/src/helpers/setupI18n.js#L27-L29

We also expose the i18next object on `window.readerControl`, which means you can access through `instance` in your app. This means you could use [addResourceBundle](https://www.i18next.com/overview/api#addresourcebundle) to add your own resources dynamically.
https://github.com/PDFTron/webviewer-ui/blob/9da862531ab63168497a1d26b41d235d092c5e1b/src/apis/index.js#L268

To summarize, the following is what you can do in the current WebViewer version:
```javascript
WebViewer(
  {
    path: '/lib',
    disableI18n: true
  },
  document.getElementById('viewer')
).then(instance => {
  instance.i18n.addResourceBundle(
    'en',
    'translation',
    {
      action: {
        apply: 'Apply text'
      },
      // ...
      tool: {
        pan: 'Pan'
      }
    },
    true,
    true
  );
});
```

Can you try the above code to see if it works for you?

**Comment by soymanuvalle** [comment]
Thanks @ZhijieZhang for your response. I noticed the `disableI18n` option but I thought it was something pending to release as I didn't know about the `instance.i18n`.

The solution you've provided works perfectly for our environment and I think it's cleaner.

Also thanks for the explanation about `getHashParams`

**Comment by ZhijieZhang** [comment]
Nice. Glad to hear that you were able to get this working the way you wanted on your end.

Do you mind me closing the issue and the PR at this moment? In a future version we may rename the `disableI18n` option to be more descriptive and intuitive(and document `instance.i18n`).
